### PR TITLE
[3.x] GLES2: Make GPU skinning more consistent with GLES3

### PR DESCRIPTION
**Pull request by Chlorobite**
My first contribution here! 👋

Aside from consistency, functionally, this may improve performance, with a single ivec4 cast as opposed to 4 separate int casts for bone_ids, plus calling transpose() 1 time instead of 4.

But this change is mainly motivated by the vec4 indexing in the previous implementation causing the following issue on my [Xiaomi Redmi 6A](https://www.gsmarena.com/xiaomi_redmi_6a-9217.php) device on WebGL (a desktop and a high-end mobile device, and a native export on the same device worked, so this is entirely a driver+ANGLE issue):
Screenshot_2023-08-15-11-07-43-365_com android chrome

With the GLES3 implementation ported over, the model renders correctly (the issue with her clothes is common in Makehuman, the rendering is technically correct):
Screenshot_2023-08-15-11-08-19-569_com android chrome

The minimal fix for the issue would be changing
`bone_transform += transpose(b) * bone_weights[i];`
to:
`bone_transform += transpose(b) * dot(bone_weights, vec4(i == 0 ? 1.0 : 0.0, i == 1 ? 1.0 : 0.0, i == 2 ? 1.0 : 0.0, i == 3 ? 1.0 : 0.0));`
but this would degrade performance, and looks quite ugly.

### CODE REVIEW

**Comment by Benedani** [comment]
Eh, sure, if it needs to be *that* indented...

File: drivers/gles2/shaders/scene.glsl [hunk 1]
```diff
@@ -433,17 +433,44 @@ void main() {
...
 
-			highp mat4 b = mat4(
-					texel2DFetch(bone_transforms, skeleton_texture_size, tex_ofs + ivec2(0, 0)),
+		ivec2 tex_ofs = ivec2(bone_indicesi.x * 3, 0);
+		bone_transform = mat4(
+					texel2DFetch(bone_transforms, skeleton_texture_size, tex_ofs),
 					texel2DFetch(bone_transforms, skeleton_texture_size, tex_ofs + ivec2(1, 0)),
 					texel2DFetch(bone_transforms, skeleton_texture_size, tex_ofs + ivec2(2, 0)),
-					vec4(0.0, 0.0, 0.0, 1.0));
+					vec4(0.0, 0.0, 0.0, 1.0)) *
```
Comment (lawnjelly) [inline drivers/gles2/shaders/scene.glsl]: Is this the same as `vec4(0.0, 0.0, 0.0, bone_weights.x)`? (or maybe you were thinking the way you have done would be cheaper?)

**Review by Chlorobite — COMMENTED** [review]
...huh? GitHub?

**Comment by lawnjelly** [comment]
Looks fine to me if it is more widely compatible and reduces the number of transposes, the commits will need squashing though and @clayjohn to take a look.

Would be nice to see some performance benchmarks especially on Android just to double check there is no regression with speed (it should in theory be faster, but might depend on how well / whether the old version gets optimized).

**Review by clayjohn — APPROVED** [review]
Looks fine to me too!

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**